go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -3811,6 +3853,69 @@ func (s *VtctldServer) ValidateVSchema(ctx context.Context, req *vtctldatapb.Val
 	return &resp, nil
 }
 
+// getTopologyCell gets a cell and its children recursively, given a cell's path
+func (s *VtctldServer) getTopologyCell(ctx context.Context, cellPath string) (*vtctldatapb.TopologyCell, error) {
+	// Extract cell and relative path
+	parts := strings.Split(cellPath, "/")
+	if parts[0] != "" || len(parts) < 2 {
+		return nil, fmt.Errorf("Invalid path: %s", cellPath)
+	}
+	cell := parts[1]
+	relativePath := cellPath[len(cell)+1:]
+	topoCell := vtctldatapb.TopologyCell{Name: parts[len(parts)-1]}
+
+	conn, err := s.ts.ConnForCell(ctx, cell)
+	if err != nil {
+		return nil, fmt.Errorf("Error connecting to cell %s: %v", cell, err)
+	}
+
+	data, _, noData := conn.Get(ctx, relativePath)
```

Comment:
this does not match the usages of `conn.Get` i see in `package topo` (which treat the third value as an error, which it is, and not a signal of "no data", which it is not)

Reply:
https://github.com/vitessio/vitess/blob/2b5f72c5dd0411c3ff9da4517c4a8d32c4f57fc5/go/vt/vtctld/explorer.go#L87 This is taken directly from the old topology browser code. noData is just a naming convention, I can rename it to err if that makes it clearer.

Reply:
i think we should improve it, rather than just copy. we should follow the other vtadmin-api endpoints and push errors back to the web ui for rendering there

Reply:
I agree! Just acknowledging it is from prior art. We can surface this error, definitely.

Reply:
yeah for sure! i think the original topo browser is _preeeeetty_ old code so a lot of the patterns and conventions have been superseded since then, so we should feel free to be pretty liberal with our port